### PR TITLE
Implement `percent_rank()` on GPU

### LINKED ISSUE #4004: [FEA] Add support for percent_rank
**Is your feature request related to a problem? Please describe.**
This is a feature quest to add support for percent_rank in windowing.

Here is a mini example:
```
val querytext="""SELECT gender,percent_rank(salary) OVER (PARTITION BY gender ORDER BY salary) from df2"""
sql(querytext).collect
```

### PR DESCRIPTION
**Pull request by NVnavkumar**
Fixes #4004.

This implements `percent_rank()` on the GPU using CUDF's existing `percentRank()` aggregation. However, it does not use the existing batch optimization code because of the non-trivial nature of that potential implementation, combined with the possibility that CUDF will change `percentRank()` in the future.  Tests have been updated as well to include separate tests for `percent_rank()` since they do not use the same optimization, so the check for `GpuRunningWindowExec` is removed in those cases.

I also added decimal 128 support to many of the tests, as this was overlooked in the past since now decimal 128 is supported in partition by.

Signed-off-by: Navin Kumar <navink@nvidia.com>

### CODE REVIEW

File: integration_tests/src/main/python/window_function_test.py [hunk 1]
```diff
@@ -473,14 +496,33 @@ def test_window_running_rank(data_gen):
         validate_execs_in_gpu_plan = ['GpuRunningWindowExec'],
         conf = conf)
 
+# Percent rank is a running window aggregation that currently does not use the Batch Optimiz
+@ignore_order(local=True)
+@pytest.mark.parametrize('data_gen', all_basic_gens + [decimal_gen_32bit, decimal_gen_128bit], ids=idfn)
+def test_window_running_percent_rank(data_gen):
```
Comment (revans2) [inline integration_tests/src/main/python/window_function_test.py:502]: I think this is already covered by `test_window_aggs_for_rows`

**Review by revans2 — APPROVED** [review]
Looking good really just some nits

File: integration_tests/src/main/python/window_function_test.py [hunk 2]
```diff
@@ -882,6 +925,8 @@ def test_running_window_function_exec_for_all_aggs():
             (partition by a order by b,c_int) as rank_val,
           dense_rank() over
             (partition by a order by b,c_int) as dense_rank_val,
+          percent_rank() over
```
[the remark this answers is not in the document]
Comment (NVnavkumar) [reply integration_tests/src/main/python/window_function_test.py]: Test fixes covered in latest commit

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuWindowExpression.scala [hunk 3]
```diff
@@ -1526,3 +1526,37 @@ case class GpuLag(input: Expression, offset: Expression, default: Expression)
     }
   }
 }
+
+/**
+ * Technically, percent_rank() *is* a running window function, but because of
```
[the remark this answers is not in the document]
Comment (NVnavkumar) [reply sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuWindowExpression.scala]: Updated comment in latest commit

**Comment by NVnavkumar** [comment]
build

**Review by revans2 — APPROVED** [review]

**Pull request merged**